CMakeLists.txt

```
@@ -429,6 +423,43 @@ else()
   set(PLUGIN_LLVM_VERSION_MAJOR 0)
 endif()
 
+set(DEFAULT_GPU_ARCH "" CACHE STRING "(Deprecated, use DEFAULT_TARGETS instead) Optional: Default GPU architecture to compile for when targeting GPUs (e.g.: sm_60 or gfx900)")
+set(DEFAULT_TARGETS "" CACHE STRING "Default targets to compile for")
+
+if(NOT DEFAULT_TARGETS)
+  if(DEFAULT_PLATFORM)  
+    message(DEPRECATION "DEFAULT_PLATFORM is deprecated; use DEFAULT_TARGETS instead.")
+    
+    if(DEFAULT_PLATFORM STREQUAL "cpu")
+      set(DEFAULT_TARGETS "omp")
+    endif()
+    
+    if(DEFAULT_GPU_ARCH)
+      message(DEPRECATION "DEFAULT_GPU_ARCH is deprecated; use DEFAULT_TARGETS instead.")
+
+      if(DEFAULT_PLATFORM STREQUAL "cuda")
+        set(DEFAULT_TARGETS "cuda:${DEFAULT_GPU_ARCH}")
+      elseif(DEFAULT_PLATFORM STREQUAL "rocm")
+        set(DEFAULT_TARGETS "hip:${DEFAULT_GPU_ARCH}")
```

If we're breaking the old behavior, let's at least be very clear about it:

```suggestion
        set(DEFAULT_TARGETS "hip:${DEFAULT_GPU_ARCH}")
      else()
        message(SEND_ERROR "Invalid value for DEFAULT_PLATFORM: \"${DEFAULT_PLATFORM}\". When DEFAULT_GPU_ARCH is specified, only \"cuda\" and \"hip\" are supported.")
```

---

We can do that. Is there any particular reason why the error message talks about `"cuda"` and `"hip"`? My understanding was that the `DEFAULT_PLATFORM` variable previously allowed "cuda" and "rocm", so shouldn't it be:

```suggestion
        set(DEFAULT_TARGETS "hip:${DEFAULT_GPU_ARCH}")
      else()
        message(SEND_ERROR "Invalid value for DEFAULT_PLATFORM: \"${DEFAULT_PLATFORM}\". When DEFAULT_GPU_ARCH is specified, only \"cuda\" and \"rocm\" are supported.")
```

---

Yes, you're right